### PR TITLE
Added ccr to xpack usage infrastructure

### LINKED ISSUE #37221: Include CCR in the xpack usage api
Relates to #30086

### PR DESCRIPTION
**Pull request by martijnvg**
Closes #37221

### CODE REVIEW

**Comment by ycombinator** [comment]
Hi folks, we have a PR in Kibana that's waiting on the functionality in this PR: https://github.com/elastic/kibana/pull/28840. Any chance we could get this reviewed soon? Thanks!

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/Ccr.java [hunk 1]
```diff
@@ -311,6 +315,14 @@ public void onIndexModule(IndexModule indexModule) {
         }
     }
 
+    public Collection<Module> createGuiceModules() {
```
Comment (dnhatn) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/Ccr.java]: nit: add override annotation.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/CCRFeatureSet.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    @Override
+    public void usage(ActionListener<XPackFeatureSet.Usage> listener) {
+        MetaData metaData = clusterService.state().metaData();
+
+        int numberOfFollowerIndices = 0;
+        long lastFollowerIndexCreationDate = 0L;
+        for (IndexMetaData imd : metaData) {
+            if (imd.getCustomData("ccr") != null) {
```
Comment (dnhatn) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/CCRFeatureSet.java]: nit: maybe use `CCR_CUSTOM_METADATA_KEY` constant instead?

**Review by dnhatn — APPROVED** [review]
I left some minor nits but LGTM.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/CCRFeatureSet.java (hunk 2 above)
Comment (martijnvg) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/CCRFeatureSet.java]: Unfortunately this constant is not available from the core module.

Comment (dnhatn) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/CCRFeatureSet.java]: ok.

**Comment by martijnvg** [comment]
@dnhatn I made a slight modification to this PR. Instead of returning `last_follower_index_creation_date`, I change the ccr feature set to return `time_since_last_index_followed`. I think this is more valuable to return than a timestamp.

**Comment by dnhatn** [comment]
@martijnvg Yes, the new stat is more valuable 💯. I am fine with `time_since_last_index_followed ` but maybe just call it the `last_follow_time`?

**Comment by martijnvg** [comment]
@dnhatn I've renamed the field to `last_follow_time_in_millis`. Also `last_follow_time_in_millis` will not be reported if there are no follower indices (the returned value wouldn't make sense).

**Review by dnhatn — APPROVED** [review]
Thanks @martijnvg.

**Pull request merged**